performance.lic

```diff
@@ -2,43 +2,54 @@
   Documentation: https://elanthipedia.play.net/Lich_script_repository#performance
 =end
 
-custom_require.call(%w[common drinfomon spellmonitor events])
+custom_require.call(%w[common common-items drinfomon spellmonitor events])
 
 class Performance
-  include DRC
 
   def initialize
     no_pause_all
     start_time = Time.new.to_i
     arg_definitions = [
       [
         { name: 'noclean', regex: /noclean/i, optional: true, description: 'Skip cleaning your instrument' },
-        { name: 'checksong', regex: /checksong/i, optional: true, description: 'Only check what song to play for best learning then exit.' }
+        { name: 'checksong', regex: /checksong/i, optional: true, description: 'Only check what song to play for best learning then exit.' },
+	{ name: 'mindstate_goal', regex: /(\d+)/, optional: true, description: 'Train until mindstate >= mindstate_goal' }
       ]
     ]
 
     args = parse_args(arg_definitions)
 
     @settings = get_settings
-    @song_list = get_data('perform').perform_options
+    song_list = get_data('perform').perform_options
+    @mindstate_goal = args.mindstate_goal.to_i
     skip_clean = args.noclean
     check_song_only = args.checksong
 
-    instrument = @settings.instrument
-    is_instrument_worn = instrument.nil?
+    worn_instrument = @settings.worn_instrument
```

Comment:
I'm confused by the number of vars we're using here. Is my understanding right?
* `worn_instrument` - string, the instrument name which can be played while worn, optional
* `@instrument` - string, the instrument name which must be gotten before playing, optional
* `is_instrument_worn` - bool, whether or not the instrument can be played while worn, required
* `@instrument_out` - bool, whether or not we got an instrument, required
* `@hand_armor` - string, the hand armor to be removed if the instrument is _not_ worn, optional
* `@hand_armor_removed` - bool, whether or not the hand armor was removed, required